spec.bs

```diff
@@ -0,0 +1,393 @@
+
+Exposing to FLEDGE {#fledge}
+============================
+
+<xmp class="idl">
+partial interface FledgeWorkletGlobalScope {
+  readonly attribute PrivateAggregation privateAggregation;
+};
+
+[Exposed=FledgeWorkletGlobalScope]
+dictionary PASignalValue {
+  required DOMString baseValue;
+  double scale;
+  (bigint or long) offset;
+};
+
+[Exposed=FledgeWorkletGlobalScope]
+dictionary PAExtendedHistogramContribution {
+  required (PASignalValue or bigint) bucket;
+  required (PASignalValue or long) value;
+};
+
+[Exposed=FledgeWorkletGlobalScope]
+partial interface PrivateAggregation {
+  undefined reportContributionForEvent(DOMString event, PAExtendedHistogramContribution contribution);
+};
+</xmp>
+
+Issue: Do we want to align naming with implementation?
+
+Immediately after an auction completes, [=process the FLEDGE contributions cache=] with the worklet's [=contributions cache=] and the worklet's global scope.
+
+Issue: Does FLEDGE have one global scope per auction or multiple? If multiple, will need to change scope.
+
+Issue: How to handle fenced frame-triggered contributions and other event-triggered contributions.
+
+<div algorithm>
+The <dfn method for="PrivateAggregation">reportContributionForEvent(DOMString event, PAExtendedHistogramContribution contribution)</dfn> method steps are:
+</div>
+
+1. Run the [=PrivateAggregation/validate an extended histogram contribution=] steps on |contribution|.
+
+Issue: Fill in the rest. (Need to put the contribution in some sort of queue and process the queue at some point. Need to decide where the queue should live given it has to outlive the auction.)
+
+To <dfn>process the FLEDGE contributions cache</dfn> given a [=contributions cache=] |contributionsCache| and a {{FledgeWorkletGlobalScope}} |scope|, run the following steps:
+1. Let |filledInContributions| be a new empty [=list=].
+1. [=list/iterate|For each=] |contribution| of |contributionsCache|:
+    1. [=list/Append=] the result of [=filling in the contribution=] |contribution| to |filledInContributions|.
+1. Run the [=PrivateAggregation/report scheduling steps=] with |scope|'s [=relevant settings object=]'s [=origin=], "<code>[=worklet type identifier/fledge=]</code>" and |filledInContributions|.
+
+
+Structures {#structures}
+========================
+
+<h3 dfn-type=dfn>Aggregatable report</h3>
+
+An aggregatable report is a [=struct=] with the following items:
+<dl dfn-for="aggregatable report">
+: <dfn>reporting endpoint</dfn>
+:: A (suitable?) URL
+: <dfn>original report time</dfn>
+:: A [=moment=]
+: <dfn>report time</dfn>
+:: A [=moment=]
+: <dfn>contributions</dfn>
+:: An [=ordered set=] of {{PAHistogramContribution}}
```

Comment:
Can we use `{{PAHistogramContributions}}`? Or `{{PAHistogramContribution}}s`?

Reply:
Went with the latter